src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -1617,4 +1652,121 @@ private static class OutputPayload implements Serializable {
             assumptionFailure = error instanceof AssumptionViolatedException ? error.getMessage() : null;
         }
     }
+
+    /**
+     * Alternative to {@link #addPlugins} or {@link TestExtension} that lets you build a test-only plugin on the fly.
+     * ({@link ExtensionList#add(Object)} can also be used for certain cases, but not if you need to define new types.)
+     */
+    public final class SyntheticPlugin {
+        private final String pkg;
+        private String shortName;
+        private String version = "1-SNAPSHOT";
+        private Map<String, String> headers = new HashMap<>();
+
+        SyntheticPlugin(String pkg) {
+            this.pkg = pkg;
+            shortName = pkg.replace('.', '-');
```

Comment:
Can be distinguished from usual plugins because the convention is different, but in my mind it would be clearer if we used a clear prefix to identify synthetic plugins (e.g. `test-<shortName>` or `synthetic-<shortName>`)